app/src/main/java/org/mozilla/fenix/settings/search/RadioSearchEngineListPreference.kt

```diff
@@ -155,6 +166,10 @@ class RadioSearchEngineListPreference @JvmOverloads constructor(
         context: Context,
         engine: SearchEngine,
     ) {
+        val nextSearchEngine = context.components.core.store.state.search.searchEngines.first {
+            it.id != engine.id && (it.isGeneral || it.type == SearchEngine.Type.CUSTOM)
+        }
+        context.components.useCases.searchUseCases.selectSearchEngine(nextSearchEngine)
```

Comment:
Thanks to @Mugurell this fixes the behavior of selecting a topic specific search engine instead of the next general/custom engine.

https://user-images.githubusercontent.com/32488956/207871980-c8ccd126-fe34-41fb-a0d5-271ee8e7e342.mp4